server/game/cards/07-GR/DeEscalation.js

```diff
@@ -0,0 +1,34 @@
+const Card = require('../../Card.js');
+
+class DeEscalation extends Card {
+    // Play: Destroy each creature. Your opponent archives the top 3
+    // cards of their deck.
+    setupCardAbilities(ability) {
+        this.play({
+            effect: 'destroy all creatures{1}',
+            effectArgs: (context) => [
+                context.player.opponent
+                    ? ' and have ' +
+                      context.player.opponent.name +
+                      ' archive the top 3 cards of their deck'
+                    : ''
+            ],
+            gameAction: ability.actions.sequential([
+                ability.actions.destroy((context) => ({ target: context.game.creaturesInPlay })),
+                ability.actions.conditional((context) => ({
+                    condition: context.player.opponent && context.player.opponent.deck.length > 0,
+                    trueGameAction: ability.actions.archive({
+                        target: context.player.opponent.deck.slice(
+                            0,
+                            Math.min(context.player.deck.length, 3)
```

Comment:
Should be context.player.opponent.deck.length

No need to worry as slice() returns less than 3 if array is small.
```
> [1,2].slice(0,3)
[ 1, 2 ]
> 
```

Reply:
Done, thanks.